### ISSUE TITLE
save model for inference

### ISSUE THREAD

**Issue by Mahmuod1** [qubvel/segmentation_models.pytorch#657]
i would thank you in your talented repos such as this repo 

i want to save my custom model to load it and not causing the error segmentaiton_models.pytorch not found 
and not installaing the pakage what would i have to do save the model class with the model_state_dict or what 
i was able to do this with the tensorflow repo compiling the model with its loss and meterices and load the model as any tensorflow model doesn't need to install any thing just tensorflow framework
how i can do the same with this pytorch model

**Comment by t110368027** [comment]
hi @Mahmuod1 

here is a simple code to answer from your question

```python=3.7
import segmentation_models_pytorch as smp
import torch

model = smp.Unet(
    encoder_name="resnet34",        # choose encoder, e.g. mobilenet_v2 or efficientnet-b7
    encoder_weights="imagenet",     # use `imagenet` pre-trained weights for encoder initialization
    in_channels=1,                  # model input channels (1 for gray-scale images, 3 for RGB, etc.)
    classes=3,                      # model output channels (number of classes in your dataset)
)
torch.save(model, "save model path")
```
```python=3.7
import torch
model = torch.load("save model path")
```

**Comment by Mahmuod1** [comment]
thanks for your responding 
but i mean to load the trained model without install any package just pytorch
